src/web-mercator-viewport.js

```diff
@@ -221,3 +226,52 @@ export function fitBounds({
     zoom
   };
 }
+
+// Apply mathematical constraints to viewport props
+/* eslint-disable complexity */
+export function normalizeViewportProps({
```

Comment:
Should it be its own file? Strange to have a big util in a file whose main intent is to define a class...